holopy/inference/tests/test_prior.py

```diff
@@ -87,59 +347,167 @@ def test_scale_factor():
+
+    @attr("fast")
+    def test_prior_array_math(self):
+        expected_sum = np.array([Gaussian(1, 2), Gaussian(2, 2)])
+        expected_product = np.array([Gaussian(1, 2), Gaussian(2, 4)])
+        self.assertTrue(np.all(self.g + np.array([0, 1]) == expected_sum))
+        self.assertTrue(np.all(self.g * np.array([1, 2]) == expected_product))
+
+    @attr("fast")
+    def test_invalid_math(self):
+        with self.assertRaises(TypeError):
+            self.u + self.u
+        with self.assertRaises(TypeError):
+            self.g + self.b
+        with self.assertRaises(TypeError):
+            self.g + [0,1]
+        with self.assertRaises(TypeError):
+            self.c + self.u
+        with self.assertRaises(TypeError):
+            self.c + self.g
+        with self.assertRaises(TypeError):
+            self.c * (1 + 1j)
+        with self.assertRaises(TypeError):
+            self.g * self.g
 
 def test_generate_guess():
     gold1 = np.array([[-0.091949, 0.270532], [-1.463350, 0.691041],
         [1.081791, 0.220404], [-0.239325, 0.811950], [-0.491129, 0.010526]])
     gold2 = np.array([[-0.045974, 0.535266], [-0.731675, 0.745520],
         [0.540895, 0.510202], [-0.119662, 0.805975], [-0.245564, 0.405263]])
     pars = [Gaussian(0,1), Uniform(0,1,0.8)]
-    guess1 = prior.generate_guess(pars, 5, seed=22)
-    guess2 = prior.generate_guess(pars, 5, scaling=0.5, seed=22)
+    guess1 = generate_guess(pars, 5, seed=22)
+    guess2 = generate_guess(pars, 5, scaling=0.5, seed=22)
     assert_allclose(guess1, gold1, atol=1e-5)
     assert_allclose(guess2, gold2, atol=1e-5)
 
+class TestMakeCenterPriors(unittest.TestCase):
+    @property
+    def image(self):
+        img = np.zeros([4,4])
+        img[:3, 1:] = np.pad(np.zeros([1,1]), 1, 'constant', constant_values=1)
+        return data_grid(img, spacing=2)
+
+    @attr("fast")
+    def test_my_image(self):
+        self.assertTrue(np.allclose(self.image.values, np.array(
+            [[0,1,1,1],[0,1,0,1],[0,1,1,1],[0,0,0,0]])))
+
+    @attr("fast")
+    def test_basic(self):
+        expected = [Gaussian(0, 2), Gaussian(2, 2), Uniform(0, 40)]
+        evaluated = make_center_priors(self.image)
+        self.assertEqual(evaluated, expected)
+
+    def test_z_range_extents(self):
+        expected = [Gaussian(0, 2), Gaussian(2, 2), Uniform(0, 16)]
+        evaluated = make_center_priors(self.image, z_range_extents = 2)
+        self.assertEqual(evaluated, expected)
+
+    def test_xy_uncertainty(self):
+        expected = [Gaussian(0, 4), Gaussian(2, 4), Uniform(0, 40)]
+        evaluated = make_center_priors(self.image, xy_uncertainty_pixels = 2)
+        self.assertEqual(evaluated, expected)
+
+    def test_z_range_units(self):
+        expected = [Gaussian(0, 2), Gaussian(2, 2), Uniform(2, 10)]
+        evaluated = make_center_priors(self.image, z_range_units=(2, 10))
+        self.assertEqual(evaluated, expected)
```

Comment:
Oops one more comment. The end of each test file should include the 2 lines:
```python
if __name__ == '__main__':
    unittest.main()
```
so they can be run from the command line individually. Nosetests is good at sniffing these out, so it doesn't matter so much, but still. We can chat if you want.